File: PSP-DAM-ACTEVA03B/src/cocina/Cocinero.java
package cocina;

class Cocinero extends Thread {
    private int idCocinero;
    private Cocina cocina;

    public Cocinero(int idCocinero, Cocina cocina) {
        this.idCocinero = idCocinero;
        this.cocina = cocina;
    }

    @Override
    public void run() {
        while (true) {
            Pedido pedido = cocina.tomarPedido();
            if (pedido == null) { // No quedan pedidos
                break;
            }
            prepararPedido(pedido);
        }
    }

    private void prepararPedido(Pedido pedido) {
        System.out.println("Cocinero " + idCocinero + " está preparando el pedido " + pedido.getId() + ": " + pedido.getNombrePlato());
        try {
            Thread.sleep((long) (Math.random() * 2000) + 1000); // Simula el tiempo de preparación
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        cocina.registrarPedido(pedido, idCocinero);
    }
}

File: PSP-DAM-ACTEVA03 Python/cocinero.py
import threading
import time
import random

class Cocinero(threading.Thread):
    def __init__(self, id_cocinero, cocina):
        super().__init__()
        self.id_cocinero = id_cocinero
        self.cocina = cocina

    def run(self):
        while True:
            pedido = self.cocina.tomar_pedido()
            if pedido is None:
                break
            self.preparar_pedido(pedido)

    def preparar_pedido(self, pedido):
        print(f"Cocinero {self.id_cocinero} está preparando el pedido {pedido.id}: {pedido.nombre_plato}")
        time.sleep(random.uniform(1, 3))  # Simula tiempo de preparación
        self.cocina.registrar_pedido(pedido, self.id_cocinero)
